### PR TITLE
Bring back TypeGraphQL

### PR DESCRIPTION
**Pull request by Pl217**
Most of the changes are cleanup and myself getting up to speed with the project and trying out the tools and libraries on Location data.

### CODE REVIEW

File: src/domain-services/base-types.ts [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+export class BaseType {
+  @Field()
+  createdAt: Date;
+
+  @Field()
+  updatedAt: Date;
+}
+
+@ObjectType()
+export class ParanoidType extends BaseType {
```
Comment (astroash) [inline src/domain-services/base-types.ts]: Is this concept borrowed from `sequalize`? (It's the only time I've come across soft delete being described in this way).

Personally I think that this is not descriptive naming, and would prefer something along the lines of `BaseTypeWithDeletedAt` or `BaseTypeWithSoftDelete`

File: src/domain-services/location/location-service.ts [hunk 2]
```diff
@@ -0,0 +1,23 @@
+import { Service } from 'typedi';
+import { DbModels } from '../../data-providers/postgres/models';
+import { LocationTable } from '../../data-providers/postgres/models/location';
+
+@Service()
+export class LocationService {
```
Comment (astroash) [inline src/domain-services/location/location-service.ts:6]: As we're moving back to `typegraphql`'s way of doing things, it is worth updating the file structure doc https://github.com/UN-OCHA/hpc-api/blob/develop/docs/FILE_STRUCTURE.md.

We previously were going to call our db abstractions libs, however `typedi` follows the spring structure of calling these services.

**Review by astroash — CHANGES_REQUESTED** [review]
Nice work tidying this up and adding in locations. 

Two small comments around naming conventions

File: src/domain-services/base-types.ts (hunk 1 above)
Comment (Pl217) [reply src/domain-services/base-types.ts]: Yes, I've also seen this way of naming soft deletes in `sequelize` only. But used them anyway as I've seen https://github.com/UN-OCHA/hpc_service/pull/2241 enforce the term as well.

Renamed to `BaseTypeWithSoftDelete` which is indeed better name in 1f66d00ca0fdaf564406617dbe406f451e7279fa

File: src/domain-services/location/location-service.ts (hunk 2 above)
Comment (Pl217) [reply src/domain-services/location/location-service.ts:6]: Did some changes to mention services are used with help of dependency injection from `typedi` - see ba800a8aeb877bac6f4685d85793dadffd398d62.

**Comment by Pl217** [comment]
I am thinking that I should conclude my work on bringing back TypeGraphQL and complete the rest of points inside the [ticket](https://humanitarian.atlassian.net/browse/HPC-7904) with a new PR, to prevent this one from growing too much.

This mainly means adding authentication to requests, but that will also require adding some more types and resolvers.

**Review by astroash — APPROVED** [review]
These changes look great ✨

**Pull request merged**